cg/models/demultiplex/flow_cell.py

```diff
@@ -75,21 +81,28 @@ def full_name(self) -> str:
         """Return flow cell full name."""
         return self.path.name
 
+    @property
+    def sample_sheet(self) -> SampleSheet:
+        """Return sample sheet object."""
+        if not self._sample_sheet:
+            self._sample_sheet = self._read_sample_sheet()
+        return self._sample_sheet
+
+    def _read_sample_sheet(self) -> SampleSheet:
+        if not self.sample_sheet_path_hk:
+            raise FlowCellError("Sample sheet path not set")
```

Comment:
```suggestion
        if not self.sample_sheet_path_hk:
            raise FlowCellError("Sample sheet path not set")
```
Would this be needed if we set the `self.sample_sheet_path_hk` to required instead of optional? 
I think with the current flow there always has to be a sample sheet anyway.

Reply:
I thought about that too, it would not be necessary and it seems better. But `FlowCellDirectoryData` is instantiated in other parts of the code where we don't have the `sample_sheet_path`. So might require additional refactoring. 

What do you think of doing it in a separate PR?